### PR TITLE
Add FFTW builder

### LINKED ISSUE #5: fftw
https://github.com/JuliaMath/FFTWBuilder

### PR DESCRIPTION
**Pull request by ViralBShah**
Fix #5 

I copied the [build_tarballs.jl](https://github.com/JuliaMath/FFTWBuilder/blob/master/build_tarballs.jl) from the FFTW Builder repo.

### CODE REVIEW

**Comment by ViralBShah** [comment]
cc @ararslan @stevengj

**Comment by stevengj** [comment]
You might want to copy some of the other changes from JuliaMath/FFTWBuilder#1, namely disabling Neon and switching to `llvm-ranlib` on MacOS.

File: F/FFTW/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    export RANLIB=llvm-ranlib
+fi
+mkdir double && cd double
+../configure $config
+perl -pi -e "s/tools m4/m4/" Makefile # work around FFTW/fftw3#146
+make -j${nprocs}
+make install
+cd ..
+if [[ $target == powerpc64le-*  ]]; then config="$config --enable-altivec"; fi
+if [[ $target == arm-*  ]]; then config="$config --enable-neon"; fi
```
Comment (stevengj) [inline F/FFTW/build_tarballs.jl]: Might want to comment this out as well.

Comment (ViralBShah) [reply F/FFTW/build_tarballs.jl]: Comment out line 33 only?

Comment (stevengj) [reply F/FFTW/build_tarballs.jl]: yes.

**Comment by stevengj** [comment]
At some point it would be good to `--enable-avx512` on Intel as well, but last I checked the BinaryBuilder compiler didn't support the `-mavx512f` flag.

**Comment by ViralBShah** [comment]
Do the recent changes to build GMP and MPFR better also help with AVX 512? @staticfloat Should we try to use the AVX512 now, or enable in a future build?

**Comment by staticfloat** [comment]
We can request newer versions of GCC very easily within BB now, so compiler support should not be an issue.  The question is whether or not FFTW is polymorphic on CPUID or unconditionally uses AVX512; if the former, there's no problem, but if the latter, we do not yet have a seamless way for Julia to decide to use one artifact over another based on CPU capabiltiies.

**Comment by staticfloat** [comment]
I'm going to build tarballs for this version; then enable AVX512 and inspect it to figure out what it does.

**Comment by stevengj** [comment]
> The question is whether or not FFTW is polymorphic on CPUID

FFTW is polymorphic on CPUID.

**Comment by staticfloat** [comment]
Fantastic.  I'll enable everything then.

**Comment by staticfloat** [comment]
AVX512 is failing when building for MacOS due to https://github.com/FFTW/fftw3/issues/153

Switching over to `gcc` explicitly works, so we're going with that.

**Comment by ararslan** [comment]
Does that not affect FreeBSD, or are we using GCC there too?

**Comment by staticfloat** [comment]
It is a problem with `clang` so we'd want to use `GCC` on FreeBSD as well.

**Comment by staticfloat** [comment]
@stevengj you can use this as your new `build_fftw.jl` file: https://gist.github.com/staticfloat/fa9f64190f2c2234d624af3682b7da1f

**Comment by stevengj** [comment]
(I seem to recall that gcc produced faster code for FFTW anyway, when I checked it a few weeks ago.  However, if you have an AVX-512 machine you can test FFTW/fftw3#177.)

**Comment by stevengj** [comment]
@staticfloat, why was `v3.3.9-alpha1` troublesome (2814d23d58553d40ffdefae1d1afd525fc57037d)?

**Comment by staticfloat** [comment]
> why was v3.3.9-alpha1 troublesome (2814d23)?

The version itself is used to build the versions of [the JLL package](https://github.com/JuliaBinaryWrappers/FFTW_jll.jl/releases), and Pkg itself [doesn't take kindly](https://github.com/JuliaLang/Pkg.jl/pull/1350) to releases with build numbers or prerelease tags.  It's a defect that we're going to need to fix in 1.4, but for now I'm doing my best not to trigger it.

**Comment by ViralBShah** [comment]
Should FFTW.jl point to the binaries produced here (unless it is already done)?

**Comment by stevengj** [comment]
@ViralBShah, already done (https://github.com/JuliaMath/FFTW.jl/pull/105) and tagged (https://github.com/JuliaRegistries/General/pull/3297).

**Pull request merged**